OpenStack-Rabbit-Consumer/rabbit_consumer/message_consumer.py

```diff
@@ -1,281 +1,134 @@
 
-    image_name = payload.get("image_name", None)
+    image_name = rabbit_message.payload.image_name
     if not image_name:
         logger.debug("No image_name found in payload")
         return None
 
-    assert isinstance(image_name, str)
     try:
         return OsDescription.from_image_name(image_name)
     except ValueError:
         logger.debug("Image name %s does not match an Aquilon image name", image_name)
         return None
 
 
-def get_metadata_value(message, key):
-    """
-    Function which gets the value from the possible for a given metadata key
-    from the possible paths in the image or instance metadata with
-    the key in uppercase or lowercase
-    """
-    returnstring = message.get("payload").get("metadata").get(key)
-    if returnstring is None:
-        returnstring = message.get("payload").get("image_meta").get(key)
-        if returnstring is None:
-            returnstring = message.get("payload").get("metadata").get(key.lower())
-            if returnstring is None:
-                returnstring = message.get("payload").get("image_meta").get(key.lower())
-    return returnstring
-
+def consume(message: RabbitMessage):
+    if not is_aq_managed_image(message):
+        logger.info("Skipping non Aquilon managed image")
+        return
 
-def consume(message):
-    event = message.get("event_type")
-    if event == "compute.instance.create.end" and is_aq_managed_image(message):
-        _handle_create_machine(message)
+    if message.event_type == "compute.instance.create.end":
+        handle_create_machine(message)
 
-    if event == "compute.instance.delete.start":
-        _handle_machine_delete(message)
+    if message.event_type == "compute.instance.delete.start":
+        handle_machine_delete(message)
 
 
-def _handle_create_machine(message):
+def handle_create_machine(rabbit_message: RabbitMessage):
     logger.info("=== Received Aquilon VM create message ===")
+    _print_debug_logging(rabbit_message)
 
-    project_name = message.get("_context_project_name")
-    vm_id = message.get("payload").get("instance_id")
-    vm_name = message.get("payload").get("display_name")
-    username = message.get("_context_user_name")
-
-    aq_details = AqFields(
-        archetype=get_metadata_value(message, "AQ_ARCHETYPE"),
-        hostnames=convert_hostnames(message),
-        osname=get_metadata_value(message, "AQ_OS"),
-        osversion=get_metadata_value(message, "AQ_OSVERSION"),
-        personality=get_metadata_value(message, "AQ_PERSONALITY"),
-        project_id=message.get("_context_project_id"),
-    )
+    vm_data = VmData.from_message(rabbit_message)
+    os_details = is_aq_managed_image(rabbit_message)
+    network_details = openstack_api.get_server_networks(vm_data)
 
-    if not aq_details.hostnames:
+    if not network_details or not network_details[0].hostname:
+        vm_name = rabbit_message.payload.vm_name
         logger.info("Skipping novalocal only host: %s", vm_name)
         return
```

Comment:
Is the method meant to return anything?

Reply:
No, we just bail out because we can't handle a machine that's not routable